File: kicad-db-lib/src/app/part/part.ts
export class Part {
    id: number;

    // Schematic reference. E.g. "R" for resistors
    reference: string;

    // Part value. E.g. "10K". Must be unique per <see cref="Library"/>
    value: string;

    // Footprint of part. Have to match KiCad footprint libraries. E.g. "Resistor_SMD:R_0603_1608Metric"
    footprint: string;

    // Reference to symbol. ../lib1.lib:symbol1
    symbol: string;

    // Library the part have be stored in. <see cref="Value"/> must be unique per library.
    library: string;

    // Datasheet link
    datasheet: string;

    // Description of part.
    description: string;

    // Keywords to find part
    keywords: string;

    // Custom fields like manufacturer etc.
    customFields: { [name: string]: string };
}

File: src/KiCadDbLib/Models/Part.cs
using System.Collections.Generic;

namespace KiCadDbLib.Models
{
    public class Part
    {
        public int Id { get; set; }
        public string Library { get; set; } = string.Empty;
        public string Reference { get; set; } = string.Empty;
        public string Symbol { get; set; } = string.Empty;
        public string Value { get; set; } = string.Empty;
        public string Footprint { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Datasheet { get; set; } = string.Empty;
        public string Keywords { get; set; } = string.Empty;

        public IReadOnlyDictionary<string, string> CustomFields { get; set; }
            = new Dictionary<string, string>();

        public override string ToString()
        {
            return Id.ToString();
        }
    }
}